src/main/java/org/folio/list/services/refresh/ListRefreshService.java

```diff
@@ -115,12 +112,40 @@ private int importQueryResults(ListEntity list, UUID queryId) {
       list.getInProgressRefreshId().map(UUID::toString).orElse("NONE"));
     DataBatchCallback dataBatchCallback = dataBatchCallbackSupplier.get();
     int offset = 0;
-    List<List<String>> ids = queryClient.getSortedIds(queryId, offset, DEFAULT_BATCH_SIZE);
-    while (!CollectionUtils.isEmpty(ids)) {
-      offset += ids.size();
-      dataBatchCallback.accept(list, ids);
-      ids = queryClient.getSortedIds(queryId, offset, DEFAULT_BATCH_SIZE);
-    }
+    UUID entityTypeId = list.getEntityTypeId();
+    EntityType entityType = entityTypeClient.getEntityType(entityTypeId);
+    List<String> resultIdName = getIdColumnNames(entityType);
+    List<Map<String, Object>> queryContent;
+    do {
+      QueryDetails queryClientQuery = queryClient.getQuery(queryId, true, offset, DEFAULT_BATCH_SIZE);
+      queryContent = queryClientQuery.getContent();
+      List<List<String>> resultIds = queryContent.stream()
+        .map(row -> {
+          List<String> resultIdValues = new ArrayList<>();
+          for (String resultIdComponent : resultIdName) {
+            resultIdValues.add(row.get(resultIdComponent) != null ? row.get(resultIdComponent).toString() : null);
+
+          }
+          return resultIdValues;
+        })
+        .toList();
+        dataBatchCallback.accept(list, resultIds);
+        offset += queryContent.size();
+    } while (!queryContent.isEmpty());
     return offset;
   }
+  private static List<String> getIdColumnNames(EntityType entityType) {
+    return getIdColumns(entityType)
+      .stream()
+      .map(EntityTypeColumn::getName)
+      .toList();
+  }
+  private static List<EntityTypeColumn> getIdColumns(EntityType entityType) {
+    return entityType
+      .getColumns()
+      .stream()
+      .filter(column -> Boolean.TRUE.equals(column.getIsIdColumn()))
+      .sorted(Comparator.comparing(column -> column.getName().contains("tenant_id") ? 1 : 0))
+      .toList();
+  }
```

Comment:
I'd combine these
```suggestion
  private static List<String> getIdColumnNames(EntityType entityType) {
    return entityType.getColumns()
      .stream()
      .filter(column -> Boolean.TRUE.equals(column.getIsIdColumn()))
      .map(EntityTypeColumn::getName)
      .sorted()
      .toList();
  }
```
(I also simplified the sorting, since I don't think it actually matters as long as it's consistent)